src/tests/EdFi.DataManagementService.Tests.E2E/Features/Resources/UpdateResourcesValidation.feature

```diff
@@ -570,3 +582,192 @@ Feature: Resources "Update" Operation validations
+                    "endDate": "2025-03-31",
+                    "beginDate": "2025-01-01",
+                    "sessionName": "Third Quarter",
+                    "termDescriptor": "uri://ed-fi.org/TermDescriptor#Quarter",
+                    "schoolReference": {
+                        "schoolId": 4003
+                    },
+                    "totalInstructionalDays": 45,
+                    "schoolYearTypeReference": {
+                        "schoolYear": 2025
+                    }
+                  }
+                  """
+             When a POST request is made to "/ed-fi/courseOfferings/" with
+                  """
+                  {
+                    "localCourseCode": "abc",
+                    "schoolReference": {
+                        "schoolId": 4003
+                    },
+                    "sessionReference": {
+                        "schoolYear": 2025,
+                        "sessionName": "Third Quarter",
+                        "schoolId": 4003
+                    },
+                    "courseReference": {
+                        "courseCode": "ART-01",
+                        "educationOrganizationId": 4003
+                    }
+                  }
+                  """
+             Then it should respond with 201
+             # Change the sessionName
+             When a PUT request is made to "/ed-fi/sessions/{dependentId}" with
+                  """
+                  {
+                    "id": "{dependentId}",
+                    "endDate": "2025-03-31",
+                    "beginDate": "2025-01-01",
+                    "sessionName": "Fourth Quarter",
+                    "termDescriptor": "uri://ed-fi.org/TermDescriptor#Quarter",
+                    "schoolReference": {
+                        "schoolId": 4003
+                    },
+                    "totalInstructionalDays": 45,
+                    "schoolYearTypeReference": {
+                        "schoolYear": 2025
+                    }
+                  }
+                  """
+             Then it should respond with 204
+             When a GET request is made to "/ed-fi/courseOfferings/{id}"
+             Then it should respond with 200
+             # The new sessionName should cascade to this entity
+              And the response body is
+                  """
+                  {
+                    "id": "{id}",
+                    "courseReference": {
+                        "courseCode": "ART-01",
+                        "educationOrganizationId": 4003
+                    },
+                    "localCourseCode": "abc",
+                    "schoolReference": {
+                        "schoolId": 4003
+                    },
+                    "sessionReference": {
+                        "schoolId": "4003",
+                        "schoolYear": "2025",
+                        "sessionName": "Fourth Quarter"
+                    }
+                  }
+                  """
```

Comment:
How about an additional test that triggers the recursion?

Reply:
Added a test that addresses session -> courseOffering -> section.